File: 1.cpp
#include <iostream>
#include <vector>

using namespace std;

void divideAndConquer(vector<int>& items, int start, int end) {
    if (start == end) {
        cout << items[start] << " ";
        return;
    }
    
    int mid = (start + end) / 2;
    divideAndConquer(items, start, mid);
    divideAndConquer(items, mid + 1, end);
}

int main() {
    return 0;
}

File: tt/1.go
package main

import "fmt"

func printArray(A []int) {
	fmt.Println(A)
}

func divideAndConquer(A []int) {
	if len(A) <= 1 {
		printArray(A)
		return
	}
	mid := len(A) / 2
	left := A[:mid]
	right := A[mid:]
	divideAndConquer(left)
	divideAndConquer(right)
}

func main() {
	A := []int{5, 15, -30, 10, -5, 40, 10}
	divideAndConquer(A)
}
